build/helper.py

```diff
@@ -90,13 +101,19 @@ def _add_python_return_type(f):
     f['returns_python'] = f['returns']
     return f
 
-def _add_is_buffer(parameter):
-    '''Adds 'is_buffer' key/value pair to the parameter metadata iff not already populated'''
+def _add_buffer_info(parameter):
+    '''Adds buffer information to the parameter metadata iff 'size' is defined else assume not a buffer'''
     try:
-        parameter['is_buffer']
+        parameter['size']
+        parameter['is_buffer'] = True
+        parameter['ivi-dance'] = False
+        if type(parameter['size']) is str and parameter['size'].startswith('ivi-dance,'):
+            parameter['ivi-dance'] = True
+            parameter['size'] = camelcase_to_snakecase(parameter['size'].split(',')[1])
     except KeyError:
         # Not populated, assume False
         parameter['is_buffer'] = False
+        parameter['ivi-dance'] = False
```

Comment:
No, 'size' only matters when is_buffer == True.

Though we could just for completeness.